### PR TITLE
add lunar to version macro, possible fix for eol distros in status bar

### PR DESCRIPTION
**Pull request by jacquelinekay**
I believe this will hide EOL'd distros from the package pages, but I'm not sure if this is the solution I'd like to settle on.

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Some of the changes in this PR should be deployed now (e.g. updating hidden distros), some other only after Lunar has been released (e.g. active distro). Therefore those should be separated into two PRs.

**Comment by mikaelarguedas** [comment]
The lunar specific changes have been removed to be added in a follow-up, looks like I lost some changes in the process, please don't merge this just yet.

File: macro/PackageHeader.py [hunk 1]
```diff
@@ -7,14 +7,11 @@
...
-if 'cturtle' in distro_names:
-    distro_names.remove('cturtle')
-if 'diamondback' in distro_names:
-    distro_names.remove('diamondback')
-if 'unstable' in distro_names:
-    distro_names.remove('unstable')
+hidden_distros = ['boxturtle', 'cturtle', 'diamondback', 'electric', 'fuerte', 'groovy', 'hydro', 'unstable']
+
+for hidden_distro in hidden_distros if hidden_distro in distro_names:
+    distro_names.remove(hidden_distro)
```
Comment (dirk-thomas) [inline macro/PackageHeader.py]: I would suggest using list comprehension for this:

`distro_names = [d for d in distro_names if d not in hidden_distros]`

Comment (mikaelarguedas) [inline macro/PackageHeader.py]: done thanks

**Comment by dirk-thomas** [comment]
It doesn't have to be address in this PR but if you search for e.g. `boxturtle` in this repo there are numerous locations where specific distro names are being ignored. It would be good to declare this list once, like these https://github.com/ros-infrastructure/roswiki/blob/b6e22367991fb3cad82319ea08ff6e9571ba592c/macro/macroutils.py#L14-L16.

File: macro/Version.py [hunk 2]
```diff
@@ -36,19 +36,15 @@
...
 
 # configure the active set of distros
-from macroutils import distro_names as distros
+from macroutils import distro_names, distro_names_hidden
 
-if 'boxturtle' in distros:
-    distros.remove('boxturtle')
-if 'unstable' in distros:
-    distros.remove('unstable')
+distros = [d for d in distro_names if d not in distro_names_hidden]
```
Comment (dirk-thomas) [inline macro/Version.py]: I would expect this breaks the use of the `Version` macro for "hidden" distros.

**Comment by mikaelarguedas** [comment]
https://github.com/ros-infrastructure/roswiki/pull/169#issuecomment-275197540 has been addressed

File: macro/Version.py (hunk 2 above)
Comment (mikaelarguedas) [inline macro/Version.py]: Ok, I think I misunterpreted what you meant when you said 

> but if you search for e.g. boxturtle in this repo there are numerous locations where specific distro names are being ignored

could you clarify how this ties to declaring the list of hidden distros in macroutils ? do you mean creating another list for the subset of distros ignored in version?

Comment (dirk-thomas) [inline macro/Version.py]: The commit where you moved the hidden distros in a single location is good as is. The package and stack headers now use the same list of non-hidden distros.

This case is different for the version macro. It is being used within the content of wiki pages and marks e.g. a paragraph describing a certain feature with the ROS distro when it was added. So it needs to continue to work for all distros. I don't know why this should exclude "boxturtle" and "unstable" (but I wouldn't touch this).

Comment (mikaelarguedas) [inline macro/Version.py]: got it, thanks

**Comment by dirk-thomas** [comment]
The change needs to be deployed to the server after being merged.

**Comment by mikaelarguedas** [comment]
had to keep the list alteration of distro_name for banner to work properly ae0c422471a743a3c07a9fd19b130c39c0708389

**Pull request merged**